libs/barista-components/sunburst/src/sunburst.util.ts

```diff
@@ -0,0 +1,466 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DtColors,
+  DT_CHART_COLOR_PALETTE_ORDERED,
+} from '@dynatrace/barista-components/theming';
+import * as d3 from 'd3-shape';
+
+export interface DtSunburstNode {
+  label: string;
+  value?: number;
+  color?: DtColors | string;
+  children?: DtSunburstNode[];
+}
+
+export enum DtSunburstValueMode {
+  ABSOLUTE = 'absolute',
+  PERCENT = 'percent',
+}
+
+export interface DtSunburstNodeInternal {
+  origin: DtSunburstNode;
+
+  id: string;
+  label: string;
+  value: number;
+  valueRelative: number;
+  children?: DtSunburstNodeInternal[];
+
+  depth: number;
+  color: DtColors | string;
+  colorHover: DtColors | string;
+  isCurrent: boolean;
+  visible: boolean;
+  active: boolean;
+  showLabel: boolean;
+}
+
+export interface DtSunburstSlice
+  extends d3.PieArcDatum<DtSunburstNodeInternal> {
+  path: string | null;
+  labelPosition: [number, number];
+  tooltipPosition: [number, number];
+  showLabel: boolean;
+}
+
+const SVG_SETTINGS = {
+  outerRadius: 160,
+  innerRadius: 64,
+  labelOffsetRadius: 32,
+  tooltipOffsetRadius: 64,
+  minAngleForLabel: (15 / 360) * 2 * Math.PI,
+};
+const IS_SEPARATOR = '.';
+
+/*
+ *
+ *  NODE PARSING
+ *
+ */
+
+/**
+ * @description Fill the nodes with the missing information: id, depth, value (if not provided but has children), color
+ *
+ * @param nodes whole set of nodes provided by user
+ * @param parent parent of current nodes. Used id
+ * @param color color of the parent
+ */
+export const fillNodes = (nodes: DtSunburstNode[]): DtSunburstNodeInternal[] =>
+  nodes
+    .map(fillUpAndSortNodes)
+    .sort((a, b) => a.value - b.value)
+    .map((child, i, children) =>
+      fillDownNodes(child, i, children, getValue(children)),
+    );
+
+/**
+ * @description Fill value and depth of node based on it's children
+ *
+ * @param node node to fill
+ * @returns partial DtSunburstNodeInternal with all children's values and depths
+ */
+const fillUpAndSortNodes = (node: DtSunburstNode) => {
+  const children = node.children
+    ?.map(fillUpAndSortNodes)
+    .sort((a, b) => a.value - b.value);
+
+  return {
+    origin: node,
+    label: node.label,
+    children,
+    value: children ? getValue(children) : node.value ?? 0,
+    depth: children
+      ? 1 + Math.max(...children?.map(child => child.depth ?? 0))
+      : 1,
+  };
+};
+
+/**
+ * @description Fill children's ids, color and relativeValue based on the parent one
+ *
+ * @param node node to fill
+ * @param i sibling index
+ * @param nodes all siblings
+ * @param parent node to whom it belongs
+ */
```

Comment:
Very nice job with the parameter documentation. It seems that there are some missing though
```suggestion
/**
 * @description Fill children's ids, color and relativeValue based on the parent one
 *
 * @param node node to fill
 * @param i sibling index
 * @param nodes all siblings
 * @param totalValue Total sum of all values
 * @param parent node to whom it belongs
 */
```